src/test/javascript/spec/component/course-registration/course-registration-detail/course-registration-detail.component.spec.ts

```diff
@@ -56,42 +54,69 @@ describe('CourseRegistrationDetailComponent', () => {
                 fixture = TestBed.createComponent(CourseRegistrationDetailComponent);
                 component = fixture.componentInstance;
                 courseService = TestBed.inject(CourseManagementService);
+
+                // by default, assume that the course is not fully accessible but only available for registration
+                jest.spyOn(courseService, 'findOneForRegistration').mockReturnValue(of(new HttpResponse<Course>({ body: course1 })));
+                jest.spyOn(courseService, 'findOneForDashboard').mockReturnValue(throwError(new HttpErrorResponse({ status: 403 })));
             });
     });
 
     afterEach(() => {
         jest.restoreAllMocks();
     });
 
-    it('should initialize', () => {
-        component.ngOnInit();
-    });
-
     it('should parse the courseId from the route', () => {
         component.ngOnInit();
         expect(component.courseId).toBe(123);
     });
 
-    it('should load the course using findOneToRegister', fakeAsync(() => {
-        jest.spyOn(courseService, 'findOneToRegister').mockReturnValue(of(new HttpResponse<Course>({ body: course1 })));
-
+    it('should load the course using findOneForRegistration', fakeAsync(() => {
         component.ngOnInit();
 
-        expect(component.course).not.toBeNull();
+        expect(component.course).toBeDefined();
         expect(component.course?.id).toBe(course1.id);
         expect(component.course?.title).toBe(course1.title);
     }));
 
-    it('should redirect to the course page if the API endpoint redirected to /courses/:courseId/for-dashboard', async () => {
+    it('should have a function isCourseFullyAccessible that returns true if the for-dashboard endpoint returns a 200', fakeAsync(() => {
         const httpResponseComingFromForDashboardEndpoint = new HttpResponse({
             body: course1,
             headers: new HttpHeaders(),
-            url: `/api/courses/${course1.id}/for-dashboard`,
         });
-        jest.spyOn(courseService, 'findOneToRegister').mockReturnValue(of(httpResponseComingFromForDashboardEndpoint));
+        jest.spyOn(courseService, 'findOneForDashboard').mockReturnValue(of(httpResponseComingFromForDashboardEndpoint));
+
+        component.ngOnInit();
 
-        await component.ngOnInit();
+        return expect(component.isCourseFullyAccessible()).resolves.toBeTrue();
```

Comment:
I wasn't, but that's very helpful to know - thanks! I've rewritten some parts where I was working around that.

One problem I sometimes seem to get when rewriting tests using `tick()` is this:

> FATAL ERROR: Reached heap limit Allocation failed - JavaScript heap out of memory

Do you know anything about this? I cannot rewrite some of the tests because of this. Without `tick` they work flawlessly. 
I found this issue: https://github.com/facebook/jest/issues/9980 ... but neither clearing my cache nor reducing the number of workers helped solve the issue. 

For now, I'll only rewrite everything where the memory problem does not appear for me.

Reply:
Oh, one more problem I have with this is that I now get:

> ESLint: Async assertions must be awaited or returned.(jest/valid-expect)

on this line. Do you have any suggestions on how to avoid that?

Reply:
I now used `tick` and `fixture.detectChanges()` where possible and left it where I couldn't get it to work with that at all. If you find some more changes with those methods where the client tests still work after the change, feel free to suggest those changes as well